### ISSUE TITLE
Building error: tb latest version

### ISSUE THREAD

**Issue by kassane** [ZetaNumbers/tigerbeetle-rs#9]
rust: 1.72.1
OS: ArchLinux
tigerbeetle: main branch

<details>
<summary>Diff - update</summary>

```diff
diff --git a/core/src/account.rs b/core/src/account.rs
index 0efe4ef..2a904da 100644
--- a/core/src/account.rs
+++ b/core/src/account.rs
@@ -53,13 +53,13 @@ impl Account {
     }
 
     pub const fn user_data(&self) -> u128 {
-        self.0.user_data
+        self.0.user_data_128
     }
     pub fn set_user_data(&mut self, user_data: u128) {
-        self.0.user_data = user_data;
+        self.0.user_data_128 = user_data;
     }
     pub const fn with_user_data(mut self, user_data: u128) -> Self {
-        self.0.user_data = user_data;
+        self.0.user_data_128 = user_data;
         self
     }
 
@@ -102,16 +102,16 @@ impl Account {
         self
     }
 
-    pub const fn debits_pending(&self) -> u64 {
+    pub const fn debits_pending(&self) -> u128 {
         self.0.debits_pending
     }
-    pub const fn debits_posted(&self) -> u64 {
+    pub const fn debits_posted(&self) -> u128 {
         self.0.debits_posted
     }
-    pub const fn credits_pending(&self) -> u64 {
+    pub const fn credits_pending(&self) -> u128 {
         self.0.credits_pending
     }
-    pub const fn credits_posted(&self) -> u64 {
+    pub const fn credits_posted(&self) -> u128 {
         self.0.credits_posted
     }
 
diff --git a/core/src/transfer.rs b/core/src/transfer.rs
index be119ba..f3274b7 100644
--- a/core/src/transfer.rs
+++ b/core/src/transfer.rs
@@ -72,13 +72,13 @@ impl Transfer {
     }
 
     pub const fn user_data(&self) -> u128 {
-        self.0.user_data
+        self.0.user_data_128
     }
     pub fn set_user_data(&mut self, user_data: u128) {
-        self.0.user_data = user_data;
+        self.0.user_data_128 = user_data;
     }
     pub const fn with_user_data(mut self, user_data: u128) -> Self {
-        self.0.user_data = user_data;
+        self.0.user_data_128 = user_data;
         self
     }
 
@@ -127,7 +127,7 @@ impl Transfer {
     }
 
     pub const fn timeout(&self) -> Duration {
-        Duration::from_nanos(self.0.timeout)
+        Duration::from_nanos(self.0.timeout as  u64)
     }
     #[track_caller]
     pub fn set_timeout(&mut self, timeout: Duration) {
@@ -139,13 +139,13 @@ impl Transfer {
         self
     }
 
-    pub const fn amount(&self) -> u64 {
+    pub const fn amount(&self) -> u128 {
         self.0.amount
     }
-    pub fn set_amount(&mut self, amount: u64) {
+    pub fn set_amount(&mut self, amount: u128) {
         self.0.amount = amount;
     }
-    pub const fn with_amount(mut self, amount: u64) -> Self {
+    pub const fn with_amount(mut self, amount: u128) -> Self {
         self.0.amount = amount;
         self
     }
diff --git a/sys/build.rs b/sys/build.rs
index 0734b58..fdb3090 100644
--- a/sys/build.rs
+++ b/sys/build.rs
@@ -85,7 +85,7 @@ fn main() {
         )
         .arg("build")
         .arg("c_client")
-        .args((!debug).then_some("-Drelease-safe"))
+        .args((!debug).then_some("-Doptimize=ReleaseSafe"))
         .arg(format!("-Dtarget={target_lib_subdir}"))
         .current_dir(&tigerbeetle_root)
         .status()
diff --git a/sys/tigerbeetle b/sys/tigerbeetle
index 5307525..66c0f56 160000
--- a/sys/tigerbeetle
+++ b/sys/tigerbeetle
@@ -1 +1 @@
-Subproject commit 5307525e7f1a04f28c8fcb25fea9f17ce0e2a233
+Subproject commit 66c0f56eb17e30efae69f999316e571c1c078fc1
```
</details>

```bash
$> RUST_BACKTRACE=1 cargo build --workspace --examples --release
   Compiling tigerbeetle-unofficial-sys v0.2.4+0.13.123 (/home/kassane/tb_test/tigerbeetle-rs/sys)
error: failed to run custom build command for `tigerbeetle-unofficial-sys v0.2.4+0.13.123 (/home/kassane/tb_test/tigerbeetle-rs/sys)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_RELEASE_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `/home/kassane/tb_test/tigerbeetle-rs/target/release/build/tigerbeetle-unofficial-sys-1de11a5c2e16592a/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=DOCS_RS
  cargo:rerun-if-changed=src/wrapper.h
  Installing Zig 0.11.0 release build...
  /usr/bin/wget
  Downloading https://ziglang.org/download/0.11.0/zig-linux-x86_64-0.11.0.tar.xz...
  /usr/bin/wget
  Extracting zig-linux-x86_64-0.11.0.tar.xz...
  Installing zig-linux-x86_64-0.11.0 to 'zig' in current working directory...
  Congratulations, you have successfully installed Zig 0.11.0 to /home/kassane/tb_test/tigerbeetle-rs/target/release/build/tigerbeetle-unofficial-sys-5fd1abd7e9df0cc9/out/tigerbeetle/zig/zig. Enjoy!
  cargo:rustc-link-search=native=/home/kassane/tb_test/tigerbeetle-rs/target/release/build/tigerbeetle-unofficial-sys-5fd1abd7e9df0cc9/out/tigerbeetle/src/clients/c/lib/x86_64-linux-gnu
  cargo:rustc-link-lib=static=tb_client
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64-unknown-linux-gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_x86_64_unknown_linux_gnu
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=/usr/lib/clang/16/include/stddef.h
  cargo:rerun-if-changed=/usr/lib/clang/16/include/__stddef_max_align_t.h
  cargo:rerun-if-changed=/usr/lib/clang/16/include/stdint.h
  cargo:rerun-if-changed=/usr/include/stdint.h
  cargo:rerun-if-changed=/usr/include/bits/libc-header-start.h
  cargo:rerun-if-changed=/usr/include/features.h
  cargo:rerun-if-changed=/usr/include/features-time64.h
  cargo:rerun-if-changed=/usr/include/bits/wordsize.h
  cargo:rerun-if-changed=/usr/include/bits/timesize.h
  cargo:rerun-if-changed=/usr/include/bits/wordsize.h
  cargo:rerun-if-changed=/usr/include/stdc-predef.h
  cargo:rerun-if-changed=/usr/include/sys/cdefs.h
  cargo:rerun-if-changed=/usr/include/bits/wordsize.h
  cargo:rerun-if-changed=/usr/include/bits/long-double.h
  cargo:rerun-if-changed=/usr/include/gnu/stubs.h
  cargo:rerun-if-changed=/usr/include/gnu/stubs-64.h
  cargo:rerun-if-changed=/usr/include/bits/types.h
  cargo:rerun-if-changed=/usr/include/features.h
  cargo:rerun-if-changed=/usr/include/bits/wordsize.h
  cargo:rerun-if-changed=/usr/include/bits/timesize.h
  cargo:rerun-if-changed=/usr/include/bits/wordsize.h
  cargo:rerun-if-changed=/usr/include/bits/typesizes.h
  cargo:rerun-if-changed=/usr/include/bits/time64.h
  cargo:rerun-if-changed=/usr/include/bits/wchar.h
  cargo:rerun-if-changed=/usr/include/bits/wordsize.h
  cargo:rerun-if-changed=/usr/include/bits/stdint-intn.h
  cargo:rerun-if-changed=/usr/include/bits/types.h
  cargo:rerun-if-changed=/usr/include/bits/stdint-uintn.h
  cargo:rerun-if-changed=/usr/include/bits/types.h
  cargo:rerun-if-changed=/usr/lib/clang/16/include/stdbool.h

  --- stderr
  thread 'main' panicked at 'assertion failed: ch.is_ascii_uppercase()', sys/build.rs:478:9
  stack backtrace:
     0: rust_begin_unwind
               at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/std/src/panicking.rs:593:5
     1: core::panicking::panic_fmt
               at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/panicking.rs:67:14
     2: core::panicking::panic
               at /rustc/d5c2e9c342b358556da91d61ed4133f6f50fc0c3/library/core/src/panicking.rs:117:5
     3: build_script_build::screaming_snake_case_into_camel_case
     4: <build_script_build::TigerbeetleVisitor as syn::gen::visit::Visit>::visit_item_mod::{{closure}}
     5: <core::slice::iter::IterMut<T> as core::iter::traits::iterator::Iterator>::for_each
     6: <build_script_build::TigerbeetleVisitor as syn::gen::visit::Visit>::visit_item_mod
     7: syn::gen::visit::visit_item
     8: syn::gen::visit::Visit::visit_item
     9: syn::gen::visit::visit_file
    10: syn::gen::visit::Visit::visit_file
    11: build_script_build::main
    12: core::ops::function::FnOnce::call_once
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by kassane** [comment]
The build was successful after removing the assert.

**Comment by ZetaNumbers** [comment]
Solved in 4558f9e4b2f616d5e03f9f6f61818a424d2806cb
